lib/views/directory-select.js

```diff
@@ -45,15 +45,13 @@ export default class DirectorySelect extends React.Component {
     );
   }
 
-  chooseDirectory = () => new Promise(resolve => {
+  chooseDirectory = () =>
     this.props.showOpenDialog(this.props.currentWindow, {
       defaultPath: this.props.buffer.getText(),
       properties: ['openDirectory', 'createDirectory', 'promptToCreate'],
-    }, filePaths => {
-      if (filePaths !== undefined) {
+    }).then(({filePaths}) => {
```

Comment:
I think I'd prefer to use the `async`/`await` style to call the Promise-based APIs here for consistency, if you don't mind 🙇🏻 .

Also, it looks like the return value of `showOpenDialog` has changed, too; it now returns an `Object` with certain properties:

https://github.com/electron/electron/blob/v9.4.3/docs/api/dialog.md#dialogshowopendialogbrowserwindow-options

So I think this call would be something like:

```js
chooseDirectory = async () => {
  const {canceled, filePaths} = await this.props.showOpenDialog(this.props.currentWindow, {
      defaultPath: this.props.buffer.getText(),
      properties: ['openDirectory', 'createDirectory', 'promptToCreate'],
  };
  if (!canceled && filePaths.length > 0) {
    this.props.buffer.setText(filePaths[0]);
  }
}
```

But, we have to make sure that the different return value is compatible with the call signature on the Electron versions back to Atom 1.55 (the current beta), too, so we don't break our ability to backport 🤔

Reply:
Hah! I totally missed at first read that you'd already added the `{filePaths}` destructuring in your `.then` callback argument.